flax/linen/summary.py

```diff
@@ -254,21 +287,139 @@ def __call__(self, x):
 
   def _tabulate_fn(*fn_args, **fn_kwargs):
     table_fn = _get_module_table(
-        module, depth=depth, show_repeated=show_repeated
+        module,
+        depth=depth,
+        show_repeated=show_repeated,
+        compute_flops=compute_flops,
+        compute_vjp_flops=compute_vjp_flops,
     )
+
     table = table_fn(rngs, *fn_args, mutable=mutable, **fn_kwargs, **kwargs)
-    return _render_table(table, console_kwargs, table_kwargs, column_kwargs)
+
+    non_param_cols = [
+        'path',
+        'module',
+        'inputs',
+        'outputs',
+    ]
+
+    if compute_flops:
+      non_param_cols.append('flops')
+    if compute_vjp_flops:
+      non_param_cols.append('vjp_flops')
+
+    return _render_table(
+        table, console_kwargs, table_kwargs, column_kwargs, non_param_cols
+    )
 
   return _tabulate_fn
 
 
+def _get_flops(fn, *args, **kwargs):
+  e = jax.jit(fn).lower(*args, **kwargs)
+  cost = e.cost_analysis()
+  flops = int(cost['flops']) if 'flops' in cost else 0
+  return flops
+
+
+def _get_call_flops(
+    c: module_lib._CallInfo,
+    compute_flops: bool,
+    compute_vjp_flops: bool,
+) -> tuple[int, int]:
+  """Return the FLOPs of executing the call `c` in the call stack.
+
+  Does not perform actual computation / compilation / memory allocation, but
+  still introduces overhead for large modules.
+
+  Args:
+    c: ``_CallInfo``.
+    compute_flops: whether to compute forward pass FLOPs. Return `-1` otherwise.
+    compute_vjp_flops: whether to compute backward pass FLOPs. Return `-1`
+      otherwise.
+
+  Returns:
+    FLOPs of executing forward pass of `c`, and its VJP.
+  """
+
+  if not compute_flops and not compute_vjp_flops:
+    return -1, -1
+
+  rngs = jax.tree_map(
+      lambda x: x.rng, c.rngs, is_leaf=lambda x: isinstance(x, LazyRng)
```

Comment:
Why do we have `LazyRng` here?

Reply:
This is in response to https://github.com/google/flax/pull/3345#discussion_r1328870530 - I infer the rngs passed to the given submodule from its `module.scope.rngs`, which are stored as `LazyRng`.